Sources/Plasma/FeatureLib/pfConditional/plObjectInBoxConditionalObject.h

```diff
@@ -76,56 +77,61 @@ class plVolumeSensorConditionalObject : public plConditionalObject
 {
 
 protected:
-
-    hsTArray<plKey> fInside;
+    std::set<plKey>     fInside;
     int                 fTrigNum;
     int                 fType;
     bool                fFirst;
     bool                fTriggered;
-    bool                fIgnoreExtraEnters;
-public:
 
-    static bool makeBriceHappyVar;
+    plKey               fHittee;
+    uint32_t            fFlags;
 
+    enum
+    {
+        /** */
+        kIgnoreExtraEnters = (1<<0),
+        kNoServerArbitration = (1<<1),
+    };
+
+    bool IIsLocal(const plKey& key) const;
 
+public:
     enum
     {
         kTypeEnter  = 1,
         kTypeExit,
     };
+
     plVolumeSensorConditionalObject();
     ~plVolumeSensorConditionalObject(){;}
+
+    CLASSNAME_REGISTER(plVolumeSensorConditionalObject);
+    GETINTERFACE_ANY(plVolumeSensorConditionalObject, plConditionalObject);
     
-    CLASSNAME_REGISTER( plVolumeSensorConditionalObject );
-    GETINTERFACE_ANY( plVolumeSensorConditionalObject, plConditionalObject );
-    
-    virtual bool MsgReceive(plMessage* msg);
+    virtual bool MsgReceive(plMessage* msg) HS_OVERRIDE;
 
-    void Evaluate(){;}
+    void Evaluate() { }
     void Reset() { SetSatisfied(false); }
-    virtual bool Satisfied();
+    virtual bool Satisfied() HS_OVERRIDE;
     void    SetType(int i) { fType = i; }
 
     void SetTrigNum(int i) { fTrigNum = i; }
     void SetFirst(bool b) { fFirst = b; }
 
-    void IgnoreExtraEnters(bool ignore = true) {fIgnoreExtraEnters = ignore;}
+    void IgnoreExtraEnters(bool ignore = true);
+    void NoServerArbitration(bool arbitration = true);
 
-    virtual void Read(hsStream* stream, hsResMgr* mgr); 
-    virtual void Write(hsStream* stream, hsResMgr* mgr);
+    virtual void Read(hsStream* stream, hsResMgr* mgr) HS_OVERRIDE;
+    virtual void Write(hsStream* stream, hsResMgr* mgr) HS_OVERRIDE;
 
 };
 class plVolumeSensorConditionalObjectNoArbitration : public plVolumeSensorConditionalObject
```

Comment:
Should `plVolumeSensorConditionalObjectNoArbitration` set the `kNoServerArbitration` in its constructor?  Otherwise, it defaults to unset like `plVolumeSensorConditionalObject`